### DISCUSSION TITLE
关于热力图,动态对象和cesium+echarts的迁徙图要如何添加点击事件

### DISCUSSION THREAD

**Discussion by shenhubeef** [zouyaoji/vue-cesium#369] [Q&A]
关于热力图,动态对象和cesium+echarts的迁徙图要如何添加点击事件

**Comment by zouyaoji** [comment]
你的意思是要点击echarts内容吗？

不支持哈，目前并不是完美的方案，只是把echarts和cesium的容器叠在一起了 echarts的容器的`pointer-events`被设置为null了，所以拾取不了，如果不设置为null，cesium就无法正常操作了哦。 看看能不能在想要点击的地方，加一个cesium的点来点击吧。

**Comment by shenhubeef** [comment]
谢谢大神回复,那么热力图呢,我看了heatmapjs的文档也找到了获取数据的getValueAt方法但是getValueAt传入的参数好像是canvas的坐标而不是cesium里的经纬度所以不能获取正确的插值的值（热力图这个组件他不是一个实体吗）
动态对象组件给的事件似乎没有小船的change事件用于获取小船的实时经纬度
希望大神能够给我解惑




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "zouyaoji/vue-cesium"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年3月15日(星期三) 下午4:50
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [zouyaoji/vue-cesium] 关于热力图,动态对象和cesium+echarts的迁徙图要如何添加点击事件 (Discussion #369)





 
你的意思是要点击echarts内容吗？
 
不支持哈，目前并不是完美的方案，只是把echarts和cesium的容器叠在一起了 echarts的容器的pointer-events被设置为null了，所以拾取不了，如果不设置为null，cesium就无法正常操作了哦。 看看能不能在想要点击的地方，加一个cesium的点来点击吧。

**Reply by zouyaoji** [reply]
实体贴的canvas绘制的图呢，有没有看过getValueAt能满足你的需求不嘛，把heatmapInstance暴露出来你才可以调用。 

一般格点图才好这样搞把，通过经纬度获取到行列号，再去取值。热力图我也没取值过。

**Comment by shenhubeef** [comment]
getValueAt我用了它传入的参数不是Cesium的经纬度而是canvas的坐标但他本身没有点击事件获取不到canvas的坐标
听您这么说我大概明白了，这三个组件暂时是没有事件的，谢谢大神




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "zouyaoji/vue-cesium"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年3月15日(星期三) 下午5:59
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [zouyaoji/vue-cesium] 关于热力图,动态对象和cesium+echarts的迁徙图要如何添加点击事件 (Discussion #369)





 
实体贴的canvas绘制的图呢，有没有看过getValueAt能满足你的需求不嘛，把heatmapInstance暴露出来你才可以调用。
 
一般格点图才好这样搞把，通过经纬度获取到行列号，再去取值。热力图我也没取值过。
